### DISCUSSION TITLE
Interrupted VPN communication after some period of time

### DISCUSSION THREAD

**Discussion by alvarez86** [strongswan/strongswan#1785] [Q&A]
**Situation:**

We have a Next-Generation Firewall (NGFW) product, and every x days (the shortest period of time identified was 11 days, and the longest was 133), we experience a VPN issue. Our product is utilized by over 400 clients, but the problem detailed below only occurs with a few clients who have a large number of VPNs. 

**System (please complete the following information):**
**Versions:**
Kernel: Linux 5.4.113-1.el7.elrepo.x86_64 
 
StrongSwan:
strongswan-debuginfo-5.9.7-520.x86_64
strongswan-sqlite-5.9.7-520.x86_64
strongswan-tnc-imcvs-5.9.7-520.x86_64
strongswan-charon-nm-5.9.7-520.x86_64
strongswan-libipsec-5.9.7-520.x86_64
strongswan-5.9.7-520.x86_64 

**Describe the bug**
The issue is that the VPN traffic gets interrupted. 

We establish two LAN-to-LAN tunnels and, after a certain period of time, communication between the endpoints ceases. The system uses strongSwan for tunnel establishment. StrongSwan successfully installs the Security Associations (SAs) in the kernel, and everything works fine for several days.  However, at some point, the following error occurs: 

ping -I 10.165.112.248 10.10.55.1
PING 10.10.55.1 (10.10.55.1) from 10.165.112.248: 56(84) bytes of data.
ping: sendmsg: Device or resource busy
ping: sendmsg: Device or resource busy
ping: sendmsg: Device or resource busy
ping: sendmsg: Device or resource busy

When the VPN issue arises, the counter for the following parameter increases incessantly: 
watch -n1 'cat /proc/net/xfrm_stat'

XfrmOutStateProtoError (see attachment)

When the tunnel is functioning, the parameter stabilizes. 

IPs:
VPN Concentrator: 10.165.112.248
Branch: 10.10.55.1 

**To Reproduce**
As best as we know, some time after the tunnels are established, the issue appears.

**Expected behavior**
We expect the tunnels to keep running for an arbitrary amount of time.

**Logs/Backtraces**
[TCPDump_Concentrator.txt](https://github.com/strongswan/strongswan/files/11991363/TCPDump_Concentrator.txt)
[Policy.txt](https://github.com/strongswan/strongswan/files/11991365/Policy.txt)
[XFRM_Concentrator.txt](https://github.com/strongswan/strongswan/files/11991366/XFRM_Concentrator.txt)

**Additional context**
**Change of the hash protocol in the tunnels.**
When we change the encryption algorithm from MD5 to SHA1, it works, but after x days the problem reoccurs. (Not necessarily from MD5 to SHA1, but to any other). It is necessary to switch to an algorithm that is not currently in use, meaning, if we switch back to MD5, it works. However, this solution is temporary. If we switch within a short period (1 hour to the previous), the problem persists, but if we switch after several days, the problem is temporarily resolved. 

 **Server reboot.**
The problem is only resolved by restarting the server, as merely restarting the service is not sufficient. Even if after removing and reinstalling the xfrm kernel modules, the problem persists. 

 The firewall rules have already been validated and are correct.
Even with the firewall open, the problem persists, which means, with no rules in place.
The firewall policy and the frm state are okay.

**Comment by tobiasbrunner** [comment]
Looks like some kind of kernel issue. Please report it to your distro or the [netdev mailing list](http://vger.kernel.org/vger-lists.html#netdev).
